inbox/mailsync/backends/imap/common.py

```diff
@@ -43,8 +45,10 @@ def local_uids(account_id, session, folder_id, limit=None):
     if limit:
         q = q.order_by(desc(ImapUid.msg_uid))
         q = q.limit(bindparam("limit"))
-    results = q.params(account_id=account_id, folder_id=folder_id, limit=limit).all()
-    return {u for u, in results}
+    q = q.params(account_id=account_id, folder_id=folder_id, limit=limit).yield_per(
+        1000
+    )
```

Comment:
[yield_per](https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.yield_per) limits the amount of memory sqlalchemy holds at once, otherwise it will buffer everything in memory.  We buffer ourselves later to a set below. Without yield per we have two complete copies of the same data at the same time at the end of the function which increases memory pressure.